brave/src/main/java/brave/internal/recorder/MutableSpanMap.java

```diff
@@ -0,0 +1,134 @@
+package brave.internal.recorder;
+
+import brave.Clock;
+import brave.TraceContext;
+import brave.internal.Nullable;
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import zipkin.Endpoint;
+import zipkin.reporter.Reporter;
+
+/**
+ * Similar to Finagle's deadline span map, except this is GC pressure as opposed to timeout driven.
+ * This means there's no bookkeeping thread required in order to flush orphaned spans.
+ *
+ * <p>Spans are weakly referenced by their owning context. When the keys are collected, they are
+ * transferred to a queue, waiting to be reported. A call to modify any span will implicitly flush
+ * orphans to Zipkin. Spans in this state will have a "brave.flush" annotation added to them.
+ *
+ * <p>The internal implementation is derived from WeakConcurrentMap by Rafael Winterhalter. See
```

Comment:
Another small correction, it should be:
```java
@Override public boolean equals(Object other) {
  TraceContext context = get(), other = ((RealKey) other).get();
  if (context == null) {
    return other == null;
  } else {
    return context.equals(other);
  }
}
```
This way, the reference queue has a chance to clear out the reference. Alternatively, you can do:
```java
@Override public boolean equals(Object other) {
  if (this == other) {
    return true;
  }
  TraceContext context = get();  
  return context != null && context.equals(((RealKey) other).get());
}
```